dali/operators/bbox/bbox_rotate.cc

```diff
@@ -0,0 +1,391 @@
+// Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/bbox/bbox_rotate.h"
+#include "dali/core/geom/vec.h"
+
+namespace dali {
+
+DALI_SCHEMA(BBoxRotate)
+    .DocStr(
+        R"code(Transforms bounding boxes so that the boxes remain in the same place in the image after
+the image is rotated. Boxes that land outside the image with ``keep_size=True`` will be removed from the
+output if the fraction of the remaining box after cropping is less than ``remove_threshold``. If ``labels``
+are passed as a second argument, they will also be removed synchronously.
```

Comment:
`rotate` has three modes of operation - it can either extend the canvas, keep the original size or use an explicitly provided output shape. Here, the last option isn't supported. Is intentional or accidental omission?